code/Model/SiteTree.php

```diff
@@ -1604,6 +1604,21 @@ public function generateURLSegment($title)
         // Hook for extensions
         $this->extend('updateURLSegment', $t, $title);
 
+        $excludes = Director::config()->get('rules');
+        $excludes = array_keys($excludes);
+        $disallowed = array_map(function ($key) {
```

Comment:
`strto()` is pretty neat! But it fails if there's no slash in the segment, so I'm sticking to the explode :D